src/java.base/share/native/libjava/io_util.c

```diff
@@ -142,63 +146,60 @@ writeSingle(JNIEnv *env, jobject this, jint byte, jboolean append, jfieldID fid)
     }
 }
 
-void
+int
 writeBytes(JNIEnv *env, jobject this, jbyteArray bytes,
            jint off, jint len, jboolean append, jfieldID fid)
 {
-    jint n;
+    jint nwritten = -1;
     char stackBuf[BUF_SIZE];
-    char *buf = NULL;
+    char *buf = stackBuf;
     FD fd;
 
     if (IS_NULL(bytes)) {
         JNU_ThrowNullPointerException(env, NULL);
-        return;
+        return -1;
     }
 
     if (outOfBounds(env, off, len, bytes)) {
```

Comment:
we are already performing this check at Java level, do we need the perform this check again ?

Reply:
Please refer to my prior [comment](https://github.com/openjdk/jdk/pull/14981#issuecomment-1646312987).